src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -145,11 +162,47 @@ internal class RazorLanguageServerClient : ILanguageClient, ILanguageClientCusto
         var logHubLogger = _loggerProvider.CreateLogger("Razor");
         var razorLogger = new LoggerAdapter(logHubLogger);
         _server = RazorLanguageServerWrapper.Create(serverStream, serverStream, razorLogger, _projectSnapshotManagerDispatcher, ConfigureLanguageServer, _languageServerFeatureOptions);
-
+        await EnsureContainedLanguageServersInitializedAsync();
         var connection = new Connection(clientStream, clientStream);
         return connection;
     }
 
+    private async Task EnsureContainedLanguageServersInitializedAsync()
+    {
+        var clientLoadTasks = new List<Task>();
+
+#pragma warning disable CS0618 // Type or member is obsolete
+        foreach (var languageClientAndMetadata in _languageServiceBroker.LanguageClients)
+#pragma warning restore CS0618 // Type or member is obsolete
+        {
+            if (languageClientAndMetadata.Metadata is not ILanguageClientMetadata metadata)
+            {
+                continue;
+            }
+
+            if (metadata is IIsUserExperienceDisabledMetadata userExperienceDisabledMetadata &&
```

Comment:
Any implications for LiveShare?

Reply:
I've not dealt with LiveShare much so I'm not sure what the connection here is and what kind of stuff you're hoping to avoid. In terms of how the clients are selected nothing should have changed.